### PR TITLE
Add pre-commit

### PR DESCRIPTION
**Pull request by amureki**
As discussed, we want to try [pre-commit](https://pre-commit.com/#install) here.
I would not add any other linting hooks (like pydocstyle, flake8, bandit and isort) for now.
We better add them one by one along with fixing the current codebase.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+repos:
+-   repo: git@github.com:pre-commit/pre-commit-hooks
+    rev: v2.1.0
+    hooks:
+    -   id: check-added-large-files
+    -   id: debug-statements
+    -   id: end-of-file-fixer
+    -   id: name-tests-test
+    -   id: requirements-txt-fixer
+    -   id: trailing-whitespace
```
Comment (anapaulagomes) [inline .pre-commit-config.yaml]: I just tested and I only managed to get it working after removing extra spaces on it. It worked for a few hooks only:
```
WARNING] Unstaged files detected.
[INFO] Stashing unstaged files to /Users/ana.gomes/.cache/pre-commit/patch1550695946.
Check for added large files..............................................Passed
Debug Statements (Python)............................(no files to check)Skipped
Fix End of Files.........................................................Passed
Tests should end in _test.py.........................(no files to check)Skipped
Fix requirements.txt.................................(no files to check)Skipped
Trim Trailing Whitespace.................................................Passed
[INFO] Restored changes from /Users/ana.gomes/.cache/pre-commit/patch1550695946.
```
I tested with different files.

Apart from that, I'm thinking about `name-tests-test`. It appears as `Tests should end in _test.py` on `pre-commit` output, which is not accurate for this project.

Comment (amureki) [reply .pre-commit-config.yaml]: @anapaulagomes right, `name-tests-test` is not best at this stage.

Could you, please, elaborate on "get it working after removing extra spaces on it"?
"Debug statements" run whenever we have changes in any *.py file and  
"Fix requirements.txt" is only checking and sorting the order of `requirements.txt`.
I am not sure I quite got your point here. :(

Comment (amureki) [reply .pre-commit-config.yaml]: Ah, if it is only about `trailing-whitespace` check - it meant to be used for text and docstrings, whenever you have trailing      whitespaces, it notices it and removes them:

```bash
Check for added large files..............................................Passed
Debug Statements (Python)............................(no files to check)Skipped
Fix End of Files.........................................................Passed
Fix requirements.txt.................................(no files to check)Skipped
Trim Trailing Whitespace.................................................Failed
hookid: trailing-whitespace

Files were modified by this hook. Additional output:

Fixing README.rst
```

Comment (anapaulagomes) [reply .pre-commit-config.yaml]: Sorry, @amureki. On my tests, I added an `import pdb; pdb.set_trace()` in a python file and I changed `six` to `six=1` on `requirements.txt`.

Comment (amureki) [reply .pre-commit-config.yaml]: @anapaulagomes now I see. :)

I tried to add pdb in exceptions.py, it worked:
```bash
22:13 $ pre-commit
Check for added large files..............................................Passed
Debug Statements (Python)................................................Failed
hookid: debug-statements

model_mommy/exceptions.py:4:0 - pdb imported

Fix End of Files.........................................................Passed
Fix requirements.txt.................................(no files to check)Skipped
Trim Trailing Whitespace.................................................Passed
```

Which file was that? Did you stage this file so pre-commit will catch it up?

With requirements fixer - apparently it is only fixing the order, but not issues with versions:
https://pre-commit.com/hooks.html

**Pull request merged**